### PR TITLE
Loading Animation/User Input Now Required

### PR DESCRIPTION
**Pull request by DavidRamirezz**
Finished a little loading animation. Went from deciding to do a different html page for the loading animation to not because of issues with the imported packages making the information difficult to send back and forth from different pages. Decided to just remove the objects and create new ones after they are removed, much easier too.
Made the buttons have a little animation.
Added a custom style sheet as well.
Added user input requirements, although, not sure what our parameters for accepted user inputs are. I will ask about it in our next meeting. The ones right now are placeholders that can be changed if necessary. Things are commented out because I was not sure if the backend handles more inputs than just the first two airports. They do work however, to the boundaries I thought of at least.

Edit: Oh yeah, I also forgot that I changed (name="start") to (name="startAirport") and (name="Destination") to (name="destAirport") in the index file because the backend is looking for those when running. (lines 33 and 34) The NOTAMs request (from inputting two airports) was not working without making that change.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from flask import Flask, render_template, request
+from flask import Flask, render_template, request, session, redirect, url_for
```
Comment (DavidRamirezz) [inline app.py:1]: Forgot to delete this, will commit with it deleted after some people have checked the files out.

File: app.py [hunk 2]
```diff
@@ -52,11 +52,12 @@ def index():
         Notams = ParseNOTAM.ParseNOTAM(apiOutputs)
         endTime = time.time()    # Record end time
         print(f"time parsing: {endTime - startTime} seconds\n")
+
```
Comment (DavidRamirezz) [inline app.py:55]: This was used to look at the loading animation without having to call an actual request of NOTAMs (wary of the limit we have even though it is a lot.)

File: templates/index.html [hunk 3]
```diff
@@ -83,18 +116,214 @@ <h1>NOTAM Sorter</h1>
...
+    // Function to reset the body of the page and change it to loading.
+    function submitForm() {
+        // Store the form data.
+        const formData = new FormData(document.getElementById('dataForm'));
+
+        // Remvoe all the text and input boxes
+        document.getElementById('Title').remove();
+
+        document.getElementById('effectiveStartDate').remove();
+        document.querySelector('label[for="effectiveStartDate"]').remove();
```
Comment (DavidRamirezz) [inline templates/index.html:158]: Removing each element by its id seems redundant, but when trying to remove their parent element, the fetch request for 'dataForm' fails. (at least i think that is what is happening)

**Comment by JmzCS50** [comment]
LGTM

File: templates/index.html (hunk 3 above)
Comment (DavidRamirezz) [reply templates/index.html:158]: yes.

File: templates/index.html [hunk 4]
```diff
@@ -83,18 +116,214 @@ <h1>NOTAM Sorter</h1>
...
+        if (isValidStart && isValidDest && isValidDests){
+           submitForm();
+        }
+
+        //if (isValidFormat && isValidDate1 && isValidDate2 && isValidStart && isValidDest && isValidLocationRadius && isValidDests){
+        //   submitForm();
+        //}
+        
+        //submitForm();
+    }
```
[the remark this answers is not in the document]
Comment (DavidRamirezz) [reply templates/index.html:324]: Great idea, I'll do that.

**Pull request merged**